src/lib/pmd/PmdCatalogWrapper.ts

```diff
@@ -156,4 +165,51 @@ export class PmdCatalogWrapper extends PmdSupport {
   private derivePmdJarName(language: string): string {
     return path.join(PMD_LIB, "pmd-" + language + "-" + PMD_VERSION + ".jar");
   }
+
+  /**
+   * Accepts a child process created by child_process.spawn(), and a Promise's resolve and reject function.
+   * Resolves/rejects the Promise once the child process finishes.
+   * @param cp
+   * @param res
+   * @param rej
+   */
+  protected monitorChildProcess(cp: ChildProcessWithoutNullStreams, res: ([boolean, string]) => void, rej: (string) => void): void {
+    let stdout = '';
+    let stderr = '';
+
+    // When data is passed back up to us, pop it onto the appropriate string.
+    cp.stdout.on('data', data => {
+      stdout += data;
+    });
+    cp.stderr.on('data', data => {
+      stderr += data;
+    });
+
+    // When the child process exits, if it exited with a zero code we can resolve, otherwise we'll reject.
+    cp.on('exit', code => {
+      if (code === 0) {
+        this.findAndEmitEvents(stdout);
```

Comment:
I feel SfdxMessager has the scope to support exit codes as well so that we can provide more useful feedback to users. (related to my lengthy comment on `PmdRuleCataloger.java`)

Reply:
Acting on exit codes might be a bit of a stretch. I do agree that it probably makes sense to process both stderr and stdout regardless of whether we succeeded or failed, rather than putting everything in stdout. I'll make some changes and see how it goes.